### ISSUE TITLE
Documentation: RCLONE_SFTP_PASS and --sftp-pass don't mention obscure

### ISSUE THREAD

**Issue by markkutikkanen** [rclone/rclone#4252]
https://rclone.org/sftp/#sftp-pass:
```
--sftp-pass
SSH password, leave blank to use ssh-agent.

Config: pass
Env Var: RCLONE_SFTP_PASS
Type: string
Default: “”
```
When setting the password using environment variable `RCLONE_SFTP_PASS` in a script (or using the `--sftp-pass` flag) Rclone expects the password being obscured using `rclone obscure`. That is not mentioned at the page.

It is not clear where `obscure`d secrets/passwords are needed and where they aren't. It would very helpful if the fact was mentioned where applicable.

**Comment by ncw** [comment]
The config knows that variables are passwords or not so it would be quite easy to mention in here in the docs automatically for all password variables

https://rclone.org/sftp/#sftp-pass

Would you have found that?

**Comment by markkutikkanen** [comment]
If you're asking me, I don't quite understand the question. I've only been using S3 previously, which doesn't need obscuring in its secret and access keys. Now I needed SFTP and checked the documentation for its parameters and configuration. As there was no mention about obscuring, I simply put the password in the env variable and it did not work. Even the error message did not ring any bells at first. After googling the error message I found out what was wrong.

I'm sure that other than just SFTP there are other backends which need the obscured form. They all should mention this.

**Comment by ncw** [comment]
> If you're asking me, I don't quite understand the question.

Sorry for not being clear! I was noting that we generate documentation automatically for all the config variables already. Here is the `pass` for the sftp backend.

https://rclone.org/sftp/#sftp-pass

That documentation could say whether it needs to be obscured or not and I was wondering if that was sufficient documentation - whether you would have seen it?

So it would look something like

### --sftp-pass

SSH password, leave blank to use ssh-agent.

**Value must be obscured**

- Config: pass
- Env Var: RCLONE_SFTP_PASS
- Type: string
- Default: “”

I could also add something automatically to the help so it could become

      --sftp-pass string            SSH password, leave blank to use ssh-agent. (obscured)

Is either of those helpful?

**Comment by markkutikkanen** [comment]
Np, actually adding both could be helpful. Of course please add those in other storage systems requiring the obscured form as well. Thanks!

**Comment by ncw** [comment]
OK I've done this now :-)

The command line will look like

     --sftp-pass string   SSH password, leave blank to use ssh-agent. (obscured)

And the docs will look like

#### --sftp-pass

SSH password, leave blank to use ssh-agent.

**NB** Input to this must be obscured - see [rclone obscure](/commands/rclone_obscure/).

- Config:      pass
- Env Var:     RCLONE_SFTP_PASS
- Type:        string
- Default:     ""

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.52.1
